### PR TITLE
Iterable Erasure Integration

### PR DESCRIPTION
**Pull request by MarcGEthyca**
Closes #<issue>

### Description Of Changes

_Write some things here about the changes and any potential caveats_

### Code Changes

* [ ] _Erasure support for Iterable_

### Pre-PR checklist
* [X] Secrets added to 1Password
  * [X] Values for `connector_params`
  * [X] Any identity values, such as `identity_email` or `identity_phone_number`
* [X] Postman collection
  * [X] All requests in the SaaS config
  * [X] All requests to create test data
* [X] Erasure tests passing
* [x] Documentation:
  * [x] documentation complete, PR opened in fidesdocs
    https://github.com/ethyca/fidesdocs/pull/334
  * [x] documentation issue created in fidesdocs
    https://github.com/ethyca/fidesdocs/issues/335
  * [x] Connector Specification doc completed 
    https://ethyca.atlassian.net/wiki/spaces/EN/pages/2968092673/Iterable+Integration+Specification

### Steps to Confirm

* [x] Add/configure the connector in UI
* [x] Verify config and icon UI looks good 
* [ ] Create an Access DSR with the identity value
* [ ] Verify email sent by Fides with customer data is correct

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* [x] Issue Requirements are Met
* [x] Relevant Follow-Up Issues Created
* [x] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated
* [x] Update `CHANGELOG.md`

### CODE REVIEW

**Comment by MarcGEthyca** [comment]
Okay, I think I found a good spot for the note - will be happy to take any better words =)

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ The types of changes are:
 - `Security` in case of vulnerabilities.
 
 ## [Unreleased](https://github.com/ethyca/fides/compare/2.32.0...main)
+- Added Erasure support for Iterable [#4695](https://github.com/ethyca/fides/pull/4695)
```
Comment (galvana) [inline CHANGELOG.md:19]: Let's move this under the `Added` header, and lowercase `erasure`

File: data/saas/config/iterable_config.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  connector_params:
+    - name: domain
+      label: Domain
+      default_value: api.iterable.com
+      description: The Iterable API URL
+    - name: api_key
+      label: API key
+      description: The server side API key
+      sensitive: True
```
Comment (galvana) [inline data/saas/config/iterable_config.yml:17]: Nice metadata on these connector params 👌

File: tests/fixtures/saas/iterable_fixtures.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+secrets = get_secrets("iterable")
+
+
+@pytest.fixture(scope="session")
+def iterable_secrets(saas_config) -> Dict[str, Any]:
+    return {
+        "domain": pydash.get(saas_config, "iterable.domain")
+        or secrets["domain"],
+        "api_key": pydash.get(saas_config, "iterable.api_key") or  secrets["api_key"],
+        "email": pydash.get(saas_config, "iterable.email") or  secrets["email"],
```
Comment (galvana) [inline tests/fixtures/saas/iterable_fixtures.py:21]: Can you remove this `email` secret if we're not using anywhere? I might have missed where we use it, let me know!

File: tests/fixtures/saas/iterable_fixtures.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+def iterable_runner(
+    db,
+    cache,
+    iterable_secrets,
+) -> ConnectorRunner:
+    return ConnectorRunner(
+        db,
+        cache,
+        "iterable",
+        iterable_secrets,
```
Comment (galvana) [inline tests/fixtures/saas/iterable_fixtures.py:48]: Nice cleanup of the unused fixtures 👍

File: data/saas/icon/iterable.svg
Comment (galvana) [inline data/saas/icon/iterable.svg:1]: Check out the updated icon in Figma, I updated it to be consistent with our other icons.

**Review by galvana — CHANGES_REQUESTED** [review]
Just a few comments but overall this is looking good 👍

File: CHANGELOG.md (hunk 1 above)
Comment (MarcGEthyca) [reply CHANGELOG.md:19]: You got it!

File: CHANGELOG.md [hunk 5]
```diff
@@ -16,6 +16,7 @@ The types of changes are:
 - `Security` in case of vulnerabilities.
 
 ## [Unreleased](https://github.com/ethyca/fides/compare/2.33.0...main)
+- Added erasure support for Iterable [#4695](https://github.com/ethyca/fides/pull/4695)
```
Comment (galvana) [inline CHANGELOG.md:19]:
```suggestion

### Added
- Added erasure support for Iterable [#4695](https://github.com/ethyca/fides/pull/4695)
```

Comment (MarcGEthyca) [reply CHANGELOG.md:19]: I differ to you! But just fyi Kelsey suggested I moved it to unreleased cause I missed getting it merged into this release. So let me know and I'll do whichever one is best =)

**Review by galvana — APPROVED** [review]

**Pull request merged**